### PR TITLE
sprig_dodge.js tutorial

### PR DESCRIPTION
**Pull request by LucasHT22**
I accidentally deleted the source directory from PR #590

### CODE REVIEW

**Review by SamDev-7 — COMMENTED** [review]
I've made some suggestions!

File: games/sprig_dodge.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+The goal is to dodge the fireballs.
+
+Click the "open help" to discover your toolkit.
+
+--------
+Level 1
+--------
+
+Add the player and obstacles in map
```
Comment (SamDev-7) [inline games/sprig_dodge.js:26]:
```suggestion
Add the player to the map
```
Only the player is needed as fireballs are generated

File: games/sprig_dodge.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+--------
+Level 2
+--------
+
+Add controls to right, use "d" as input
```
Comment (SamDev-7) [inline games/sprig_dodge.js:32]:
```suggestion
Add controls for moving the player to right, use "d" as input
```

File: games/sprig_dodge.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+--------
+Level 3
+--------
+
+Fix the code!
+This cause this error:
+Uncaught TypeError: Cannot read properties of undefined (reading 'length') on line 153 in column 33. Open the browser console for more information.
```
Comment (SamDev-7) [inline games/sprig_dodge.js:40]:
```suggestion
Uncaught TypeError: Cannot read properties of undefined (reading 'length') on line ... in column 33. Open the browser console for more information.
```
The line probably won't be the 153 after the previous levels

File: games/sprig_dodge.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+Tip:
+the getAll() and getFirst() functions are a bit strange, aren't they?
+
+--------
+Level 4
+--------
+
+Add all functions in gameLoop
```
Comment (SamDev-7) [inline games/sprig_dodge.js:49]:
```suggestion
Add all functions in the gameLoop interval
Tip: use spawnObstacle(), moveObstacles(), despawnObstacles(), and checkHit()
```

File: games/sprig_dodge.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+--------
+END
+--------
+
+Make your own game! Try
```
Comment (SamDev-7) [inline games/sprig_dodge.js:55]:
```suggestion
Modify the game to make it your own! Try:
```

File: games/sprig_dodge.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+.....00000......
+.......0........
+.....00000......
+.......0........
+.......0........
+......0.0.......
+.....0...0......`],
+)
+
+// Map
```
Comment (SamDev-7) [inline games/sprig_dodge.js:101]:
```suggestion
// LEVEL 1 - Add player to map
```

File: games/sprig_dodge.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+var gameRunning = true; 
+
+// START - PLAYER MOVEMENT CONTROLS
+
+onInput("a", () => {
+  if (gameRunning) {
+    getFirst(player).x -= 1;
+  }
+});
+
```
Comment (SamDev-7) [inline games/sprig_dodge.js:122]:
```suggestion

// LEVEL 2 - Add move right controls
```

File: games/sprig_dodge.js [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+  for (let i = 0; i < obstacles.length; i++) {
+   if (obstacles[i].y == 8) {
+     obstacles[i].remove();
+   }
+  }
+}
+
+// See if the player was hit
+function checkHit() {
+  // fix here
```
Comment (SamDev-7) [inline games/sprig_dodge.js:154]:
```suggestion
  // LEVEL 3 - Fix code
```

File: games/sprig_dodge.js [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    if (obstacles[i].x == p.x && obstacles[i].y == p.y) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+var gameLoop = setInterval(() => {
+  // Add all functions here
```
Comment (SamDev-7) [inline games/sprig_dodge.js:168]:
```suggestion
  // LEVEL 4 - Add all game functions

**Comment by leomcelroy** [comment]
Do you feel this is ready to merge? I think we discussed this on Slack.

**Comment by LucasHT22** [comment]
For me, yes. do you have any more suggestions?

**Pull request merged**